src/core/components/hv-route/index.tsx

```diff
@@ -178,6 +212,46 @@ class HvRouteInner extends PureComponent<Types.InnerRouteProps, Types.State> {
     this.props.setPreload(id, element);
   };
 
+  /**
+   * Implement the callbacks from this class
+   */
+  updateCallbacks = (): OnUpdateCallbacks => {
+    if (!this.state.doc || !this.props.navigation) {
+      throw new NavigatorService.HvRouteError(
```

Comment:
how will this error bubble up? Is this mostly a guard for TS, or is there a risk this will be raised in which case shouldn't we try make the return value optional?

Reply:
You are correct. The doc is optional and the props.navigation is no longer necessary